src/prisma/prisma-extensions.ts

```diff
@@ -0,0 +1,32 @@
+import { Prisma, PrismaClient } from "@prisma/client";
+
+export const findUserByOAuthId = Prisma.defineExtension({
+    name: "findUserByOAuthId",
+    model: {
+        userOAuthProfile: {
+            async findUserByOAuthId(
+                providerName: string,
+                providerUserId: string,
+            ) {
+                const context = Prisma.getExtensionContext(this);
+                return context.$queryRaw`
+                            SELECT userOAuthProfile.*
+                            FROM userOAuthProfile
+                            JOIN oAuthProviders ON userOAuthProfile.providerId = oAuthProviders.id
+                            WHERE oAuthProviders.name = ${providerName} AND userOAuthProfile.providerUserId = ${providerUserId}
+                        `;
+            },
```

Comment:
I got an error from the exceptionsHandler after login to discord with my own email and password. 
![Screenshot 2024-08-11 at 6 48 27 PM](https://github.com/user-attachments/assets/d602a77d-b676-43b1-965d-6d0020eba850)